velox/expression/Expr.cpp

```diff
@@ -1126,9 +1126,17 @@ void Expr::addNulls(
   }
 
   if (result->size() < rows.end()) {
-    BaseVector::ensureWritable(
-        SelectivityVector::empty(), type, context.pool(), result);
-    result->resize(rows.end());
+    if (result->encoding() == VectorEncoding::Simple::ROW) {
+      // Avoid calling resize on all children by adding top level nulls only.
+      // We know from the check above that result is unique and isNullsWritable.
+      result->asUnchecked<RowVector>()->appendNulls(
+          rows.end() - result->size());
+    } else {
+      BaseVector::ensureWritable(
```

Comment:
I wonder why don't we add nulls by wrapping in a dictionary? CC: @bikramSingh91 @laithsakka

Reply:
this is a possible optimization as mentioned in the summary when vector is not unique.

Reply:
should we add an 'enhancement' issue to track follow ups?